common/exchange/rates.go

```diff
@@ -100,3 +100,53 @@ func CompareValue(exchangeRates common.ExchangeRates, val1 *big.Int, feeCurrency
 
 	return leftSide.Cmp(rightSide), nil
 }
+
+// RatesAndFees holds exchange rates and the basefees expressed in the rates currencies.
+type RatesAndFees struct {
+	Rates common.ExchangeRates
+
+	nativeBaseFee    *big.Int
+	currencyBaseFees map[common.Address]*big.Int
+}
+
+// NewRatesAndFees creates a new empty RatesAndFees object.
+func NewRatesAndFees(rates common.ExchangeRates, nativeBaseFee *big.Int) *RatesAndFees {
+	// While it could be made so that currency basefees are calculated on demand,
+	// the low amount of these (usually N < 20)
+	return &RatesAndFees{
+		Rates:            rates,
+		nativeBaseFee:    nativeBaseFee,
+		currencyBaseFees: make(map[common.Address]*big.Int, len(rates)),
+	}
+}
+
+// GetBaseFeeIn returns the basefee expressed in the specified currency. Returns nil
+// if the currency is not whitelisted.
+func (rf *RatesAndFees) GetBaseFeeIn(currency *common.Address) *big.Int {
+	// If native currency is being requested, return it
+	if currency == nil {
+		return rf.nativeBaseFee
+	}
+	// If a non-native currency is being requested, but it is nil,
+	// it means there is no baseFee in this context. Return nil as well.
+	if rf.nativeBaseFee == nil {
```

Comment:
Are there any downsides of always having a value for native base fee (0 by default)? That could simplify some code in other places.

Reply:
I'm following the same ordering as upstream's here, and I guess I haven't inquired enough as why it is that way.
However, in that implementation, baseFee == 0 and baseFee == nil have different behaviors:

TxA: fee 4, tip 2
TxB: fee 3, tip 3

baseFee == nil: TxA > TxB // compared by Fee first and A:4 > B:3
baseFee == 0: TxB > TxA // compared by effective tip, which is min(fee-baseFee, tip), A: 2, B: 3

why the floating priceHeap assumes nil as basefee I guess it's related to not wanting to early discard txs that we won't know what the basefee will be when they get to a block. But of course if we don't want we can have different rules here.